src/middlewared/middlewared/plugins/pool_/dataset_recordsize.py

```diff
@@ -0,0 +1,38 @@
+from middlewared.service import private, Service
+
+
+class PoolDatasetService(Service):
+
+    class Config:
+        namespace = 'pool.dataset'
+
+    # https://openzfs.github.io/openzfs-docs/Performance%20and%20Tuning/Module%20Parameters.html#zfs-max-recordsize
+    # Maximum supported (at time of writing) is 16MB.
+    MAPPING = {
+        f'{1 << 9}': '512B',
+        f'{2 << 9}': '1K',
```

Comment:
Why `2 << 9` and not `1 << 10`?
Why the mapping keys are strings?

Reply:
Because I originally had it `2 << 8` and was asked the inverse of what you asked. I'm not changing it again. I will be using `2 <<`